### PR TITLE
Add constraint method for ghsa

### PR DESCRIPTION
**Pull request by masahiro331**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -2,20 +2,17 @@ on: [push, pull_request]
...
+    runs-on: ubuntu-latest
     strategy:
       matrix:
         go-version: [1.14.x, 1.15.x]
-    runs-on: ubuntu-latest
     steps:
-    - name: Install Go
-      uses: actions/setup-go@v2
-      with:
-        go-version: ${{ matrix.go-version }}
```
Comment (knqyf263) [inline .github/workflows/test.yaml:13]: Why were they removed? They look necessary.

File: constraint.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+func (c constraint) check(v Version) bool {
+	return c.operator(v, c.version)
+}
+
+func (c constraint) String() string {
+	return c.original
+}
+
+// Returns the string format of the constraints
```
Comment (knqyf263) [inline constraint.go]:
```suggestion
// String returns the string format of the constraints
```

File: constraint.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	if err != nil {
+		return constraint{}, xerrors.Errorf("version parse error (%s): %w", m[2], err)
+	}
+	return constraint{
+		version:  v,
+		operator: constraintOperators[m[1]],
+		original: c,
+	}, nil
+}
+func (cs Constraints) Check(v Version) bool {
```
Comment (knqyf263) [inline constraint.go]:
```suggestion
}

func (cs Constraints) Check(v Version) bool {
```

File: constraint_test.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+		{">= 1.0.1.v100000", false},
+		{">= 1.1", false},
+		{">40.50.60, < 50.70", false},
+		{"2.0", false},
+		{"2.3.5-20161202202307-sha.e8fc5e5", false},
+		{">= bar", false},
+		{">= 1.1.1.v1", false},
+		{">= 1.1.1.1v", false},
+		{"==1.1.1.1v", false},
+		{"BAR >= 1.2.3", false},
```
Comment (knqyf263) [inline constraint_test.go:29]: We should have failure cases.

File: constraint_test.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+		{"> 1.0", false},
+		{"= abc", false},
+		{"> 1.0 || < foo", false},
+		{">= 1.2.3, < 2.0 || => 3.0, < 4", false},
+		{">= 1.0.1.v100000", false},
+		{">= 1.1", false},
+		{">40.50.60, < 50.70", false},
+		{"2.0", false},
+		{"2.3.5-20161202202307-sha.e8fc5e5", false},
+		{">= bar", false},
```
Comment (knqyf263) [inline constraint_test.go:25]: Is this valid for Maven?

Comment (masahiro331) [reply constraint_test.go:25]: Yes, it's valid. 
```
 $ java -jar  target/maven-artifact-4.0.0-alpha-1-SNAPSHOT.jar 40.50.60 2.0 2.3.5-20161202202307-sha.e8fc5e5 bar
Display parameters as parsed by Maven (in canonical form and as a list of tokens) and comparison result:
1. 40.50.60 -> 40.50.60; tokens: [40, 50, 60]
   40.50.60 > 2.0
2. 2.0 -> 2; tokens: [2]
   2.0 < 2.3.5-20161202202307-sha.e8fc5e5
3. 2.3.5-20161202202307-sha.e8fc5e5 -> 2.3.5-20161202202307-sha.e-8-fc-5-e-5; tokens: [2, 3, 5, [20161202202307, [sha, e, [8, [fc, [5, [e, [5]]]]]]]]
   2.3.5-20161202202307-sha.e8fc5e5 > bar
4. bar -> bar; tokens: [bar]

```

File: constraint.go (hunk 2 above)
Comment (masahiro331) [reply constraint.go]: Fixed

File: .github/workflows/test.yaml (hunk 1 above)
Comment (masahiro331) [reply .github/workflows/test.yaml:13]: Fixed.

File: go.mod [hunk 6]
```diff
@@ -1,3 +1,8 @@
 module github.com/masahiro331/go-mvn-version
 
 go 1.13
```
Comment (knqyf263) [inline go.mod]: Bump up to 1.15

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**